### DISCUSSION TITLE
How to render css-style using named resources?

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#6948] [Q&A] [answered]
I am trying to add `js` and `css` resources to my layout "using named resource". So I added the following `ResourceManifest`
```
public class ResourceManifest : IResourceManifestProvider
{
    public void BuildManifests(IResourceManifestBuilder builder)
    {
        var manifest = builder.Add();

        manifest.DefineScript("main")
                .SetUrl("~/ModuleName/js/main.min.js", "~/ModuleName/js/main.js")
                .SetVersion("1.0.0");


        manifest.DefineStyle("main")
                .SetUrl("~/ModuleName/css/main.min.css", "~/ModuleName/css/main.css")
                .SetVersion("1.0.0");
    }
}
```
Then in the layout, I added the following to the head tag
```
<style asp-name="main" asp-append-version="true"></style>
```
and the following to the bottom of the layout
```
<script asp-name="main" depends-on="jQuery" asp-append-version="true"></script>
```
The `main.js` file is being included into the html output as expected, but the `css` file isn't being rendered. What else is needed to include the `css` style into the layout?

I even tried to include the `bootstrap` and `bootstrap-select` like the following with no luck. What am I missing here?
```
<style asp-name="bootstrap" version="4.5"></style>
<style asp-name="bootstrap-select" version="1.13.18"></style>
```

**Comment by jtkech** [comment]
Did you register your `ResourceManifest`?

    public class Startup : StartupBase
    {
        public override void ConfigureServices(IServiceCollection serviceCollection)
        {
            serviceCollection.AddScoped<IResourceManifestProvider, ResourceManifest>();
        }
    }


**Update**: Hmm i assume you did it because your script has been included

So, does your layout include somewhere?

    <resources type="Stylesheet" />

**Comment by MikeAlhayek** [comment]
@jtkech  yes I did and I made sure the class gets called when the app starts. Also, please note that `<style asp-name="bootstrap"></style>` also isn't working. the `bootstrap` should be already defined by the framework.

**Comment by jtkech** [comment, marked as answer]
Okay, and do you have `<resources type="Stylesheet" />` in your theme layout?

Check your `_ViewImports.cshtml`, needs to include `@addTagHelper *, OrchardCore.ResourceManagement`

Check the project file of your current theme, needs to reference `OrchardCore.ResourceManagement`

**Comment by MikeAlhayek** [comment]
@jtkech the missed piece to the puzzle was adding `<resources type="Stylesheet" />` into my layout. I did not know that is needed along with defining the scripts to include.

Why would I need to add the `resources` tag and then list the scripts I want to use? I would think adding the `style` tag would render the scripts without having the `resources` tag.

```
    <style asp-name="bootstrap" version="4"></style>
    <style asp-name="bootstrap-select" version="1"></style>
    <style asp-name="main" asp-append-version="true" version="1"></style>
    <resources type="Stylesheet" />
```

Thanks for your help!

**Comment by jtkech** [comment]
Cool ;)

This allow to use style tags in any other view template, e.g. related to a part / field, so that when the rendering is composed, styles are enlisted through the execution of the related tag helper, and `<resources type="Stylesheet" />` in the layout is the place where all of them will be rendered.
